### PR TITLE
add POST for /api/articles/:article_id

### PR DESCRIPTION
**Pull request by lukepeteers**

### CODE REVIEW

File: __tests__/server.test.js [hunk 1]
```diff
@@ -273,8 +273,68 @@ describe('GET', () => {
...
+                    const newVotes = 5
+
+                    return request(app)
+                    .patch('/api/articles/4')
+                    .send({inc_votes: newVotes})
+                    .expect(200)
+                    .then(({body}) => {
+                        expect(body[0]).toHaveProperty('author', expect.any(String))
+                        expect(body[0]).toHaveProperty('title', expect.any(String))
+                        expect(body[0]).toHaveProperty('article_id', expect.any(Number))
```
Comment (dappernerddesigns) [inline __tests__/server.test.js:298]: You know the article id, you should be asserting it to make sure you get the correct information back.

File: __tests__/server.test.js [hunk 2]
```diff
@@ -273,8 +273,68 @@ describe('GET', () => {
...
+                        return request(app)
+                        .patch('/api/articles/999999')
+                        .send({inv_votes: 5})
+                        .expect(404)
+                        .then(({body}) =>  {
+                            const {msg} = body
+                            expect(msg).toBe('No article exists with that ID')
+                        })
+                });
+            });
```
Comment (dappernerddesigns) [inline __tests__/server.test.js:339]: You're missing a test for 400, an invalid article id on a patch.

File: controllers/controller.js [hunk 3]
```diff
@@ -53,4 +53,17 @@ exports.addComment = (request, response, next) => {
...
+}
+
+exports.getArticleToPatch = (request, response, next) => {
+    const {article_id} = request.params
+    checkArticleExists(article_id)
+    .then(() => {
+        return patchArticle(request.body, request.params)
+    })
+    .then((article) => {
+        response.status(200).send(article)
```
Comment (dappernerddesigns) [inline controllers/controller.js:65]: This should be an object with a key of article and the article requested as a value

**Review by dappernerddesigns — CHANGES_REQUESTED** [review]
This is very close but you are missing a test and have a few areas to tidy up. Resubmit when you've addressed the comments.

File: __tests__/server.test.js [hunk 4]
```diff
@@ -293,15 +293,17 @@ describe('GET', () => {
...
+                        expect(article).toHaveProperty('author', expect.any(String))
+                        expect(article).toHaveProperty('title', expect.any(String))
+                        expect(article).toHaveProperty('article_id', expect.any(Number))
+                        expect(article.article_id).toBe(4)
+                        expect(article).toHaveProperty('topic', expect.any(String))
+                        expect(article).toHaveProperty('created_at', expect.any(String))
+                        expect(article).toHaveProperty('votes', expect.any(Number))
+                        expect(article).toHaveProperty('article_img_url', expect.any(String))
+                        expect(article).toHaveProperty('votes', expect.any(Number))
+                        expect(article.votes).toBe(5)
```
Comment (Augs0) [inline __tests__/server.test.js:306]: Interesting that you assered specific info here but not for any of the other fields. Please test that the exact information we expect is coming back for all fields in line with article 2

**Review by Augs0 — APPROVED** [review]
I would update the test I commented on to be a bit more specific but otherwise you are good to merge 😄

**Pull request merged**